### ISSUE TITLE
MissingInputError under theano, but runs fine under tensorflow

### ISSUE THREAD

**Issue by billshoo** [fchollet/keras#5061]
The following Keras model runs smoothly under tensorflow, but gives "theano.gof.fg.MissingInputError" on the last line under theano (X and y can be any dummy data)

        X = np.random.rand(100000, 320)
        y = np.random.rand(100000, 1)
        model = Sequential()
        model.add(TimeDistributed(Dropout(0.5), batch_input_shape=(1, 100000, 320)))
        model.add(TimeDistributed(Dense(1000)))
        model.add(TimeDistributed((Dropout(0.5))))
        model.add(LSTM(1000, batch_input_shape=(1, 100000, 1000), 
                       return_sequences=True, stateful=True))
        model.add(TimeDistributed((Dropout(0.5))))
        model.add(TimeDistributed(Dense(1)))
        opti = RMSprop(lr=0.001)
        model.compile(optimizer=opti, loss='mse')
        model.train_on_batch(X[np.newaxis, 0:100000, :], y[np.newaxis, 0:100000])
